plan/refiner.go

```diff
@@ -128,21 +164,59 @@ func buildTableRange(p *TableScan) error {
 
 func buildSelection(p *Selection) error {
 	var err error
-	var accessConditions []expression.Expression
-	switch p.GetChildByIndex(0).(type) {
-	case *NewTableScan:
-		tableScan := p.GetChildByIndex(0).(*NewTableScan)
-		accessConditions, p.Conditions = detachConditions(p.Conditions, tableScan.Table, nil, 0)
-		err = buildNewTableRange(tableScan, accessConditions)
-		// TODO: Implement NewIndexScan
+	switch v := p.GetChildByIndex(0).(type) {
+	case *PhysicalTableScan:
+		v.AccessCondition, p.Conditions = detachTableScanConditions(p.Conditions, v.Table)
+		err = buildNewTableRange(v)
+	case *PhysicalIndexScan:
+		v.AccessCondition, p.Conditions = detachIndexScanConditions(p.Conditions, v)
+		err = buildNewIndexRange(v)
 	}
-
 	return errors.Trace(err)
 }
 
-// detachConditions distinguishs between access conditions and filter conditions from conditions.
-func detachConditions(conditions []expression.Expression, table *model.TableInfo,
-	idx *model.IndexInfo, colOffset int) ([]expression.Expression, []expression.Expression) {
+func detachIndexScanConditions(conditions []expression.Expression, indexScan *PhysicalIndexScan) (accessConds []expression.Expression, filterConds []expression.Expression) {
+	for i := len(conditions) - 1; i >= 0 && indexScan.accessEqualCount < len(indexScan.Index.Columns); i-- {
```

Comment:
We should build access condition for first column first, then second column next, the conditions need to be searched multiple times.